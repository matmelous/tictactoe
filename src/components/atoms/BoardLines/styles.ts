import styled from "styled-components";

export const Line = styled.div`
  width: 2px;
  border-radius: 10px;
  box-shadow: inset 0px 0px 2px #9ede9b, inset 0px 0px 1px #9ede9b,
    0px 0px 5px #9ede9b, inset 0px 0px 2px #000;
  height: 300px;
  background-color: #d1f5cf;
  position: absolute;

  &.line1 {
    left: 98px;
    top: 0px;
  }
  &.line2 {
    left: 198px;
    top: 0px;
  }
  &.line3 {
    left: 150px;
    top: -50px;
    transform: rotate(90deg);
  }
  &.line4 {
    left: 150px;
    top: 50px;
    transform: rotate(90deg);
  }
`;

export const Background = styled.div`
  position: absolute;
  top: 0;
  left: 0;
`;
