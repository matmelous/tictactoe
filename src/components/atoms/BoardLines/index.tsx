import React from 'react';
import { Background, Line } from './styles';

const BoardLines = () => (

  <Background>
    <Line className="line1"></Line>
    <Line className="line2"></Line>
    <Line className="line3"></Line>
    <Line className="line4"></Line>
  </Background>);

export default BoardLines;