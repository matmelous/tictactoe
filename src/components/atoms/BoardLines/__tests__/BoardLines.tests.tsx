import React from "react";
import { render, cleanup } from "@testing-library/react";
import BoardLines from "../index";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<BoardLines />);
  expect(asFragment()).toMatchSnapshot();
});