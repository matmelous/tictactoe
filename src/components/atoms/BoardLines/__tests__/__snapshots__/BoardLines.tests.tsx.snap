// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-gsDKAQ LsKEr"
  >
    <div
      class="sc-bdvvtL gMQVwe line1"
    />
    <div
      class="sc-bdvvtL gMQVwe line2"
    />
    <div
      class="sc-bdvvtL gMQVwe line3"
    />
    <div
      class="sc-bdvvtL gMQVwe line4"
    />
  </div>
</DocumentFragment>
`;
